sdk/table/azure-table/azure/table/_models.py

```diff
@@ -0,0 +1,501 @@
+    it. Or you can specify one or more of the access policy parameters within
+    the stored access policy, and the others on the URL. Finally, you can
+    specify all of the parameters on the URL. In this case, you can use the
+    stored access policy to revoke the signature, but not to modify its behavior.
+
+    Together the Shared Access Signature and the stored access policy must
+    include all fields required to authenticate the signature. If any required
+    fields are missing, the request will fail. Likewise, if a field is specified
+    both in the Shared Access Signature URL and in the stored access policy, the
+    request will fail with status code 400 (Bad Request).
+
+    :param str permission:
+        The permissions associated with the shared access signature. The
+        user is restricted to operations allowed by the permissions.
+        Required unless an id is given referencing a stored access policy
+        which contains this field. This field must be omitted if it has been
+        specified in an associated stored access policy.
+    :param expiry:
+        The time at which the shared access signature becomes invalid.
+        Required unless an id is given referencing a stored access policy
+        which contains this field. This field must be omitted if it has
+        been specified in an associated stored access policy. Azure will always
+        convert values to UTC. If a date is passed in without timezone info, it
+        is assumed to be UTC.
+    :type expiry: ~datetime.datetime or str
+    :param start:
+        The time at which the shared access signature becomes valid. If
+        omitted, start time for this call is assumed to be the time when the
+        storage service receives the request. Azure will always convert values
+        to UTC. If a date is passed in without timezone info, it is assumed to
+        be UTC.
+    :type start: ~datetime.datetime or str
+    """
+
+    def __init__(self, permission=None, expiry=None, start=None, **kwargs):  # pylint:disable=W0231
+        self.start = start
+        self.expiry = expiry
+        self.permission = permission
+
+
+class TableAnalyticsLogging(GeneratedLogging):
+
+    def __init__(  # pylint:disable=W0231
+            self,
+            **kwargs  # type: Any
+    ):
+        # type: (...)-> None
+        """Azure Analytics Logging settings.
+
+           All required parameters must be populated in order to send to Azure.
+
+           :keyword str version: Required. The version of Storage Analytics to configure.
+           :keyword bool delete: Required. Indicates whether all delete requests should be logged.
+           :keyword bool read: Required. Indicates whether all read requests should be logged.
+           :keyword bool write: Required. Indicates whether all write requests should be logged.
+           :keyword ~azure.table.RetentionPolicy retention_policy: Required.
+               The retention policy for the metrics.
```

Comment:
let's move this docstring out of init and into class. also these should be defined as :ivar: